### PR TITLE
Remove KeypairInsecureClone trait and add insecure_clone() instead

### PR DESCRIPTION
**Pull request by ckamm**
See discussion in #26248

@behzadnouri 

I didn't use `mock_clone()` because it doesn't have something to do with mocks, and the insecure wording stands out more.

### CODE REVIEW

File: sdk/src/signer/keypair.rs [hunk 1]
```diff
@@ -64,6 +64,20 @@ impl Keypair {
     pub fn secret(&self) -> &ed25519_dalek::SecretKey {
         &self.0.secret
     }
+
+    /// Allows Keypair cloning
+    ///
+    /// Note that the `Clone` trait is intentionally unimplemented because making a
+    /// second copy of sensitive secret keys in memory is usually a bad idea.
+    ///
+    /// Only use this in tests or when strictly required.
```
Comment (behzadnouri) [inline sdk/src/signer/keypair.rs:73]: Maybe add a comment here to prefer to use `Arc<Keypair>` in new code.

**Review by behzadnouri — APPROVED** [review]
I think ideally we want to replace all the call-sites with `Arc<Keypair>`.
But at least this seems to be better than the current code, and we can follow up later with `Arc<Keypair>`.

So, lgtm.

**Comment by behzadnouri** [comment]
> It looks to me like the checks are failing for unrelated reasons. Could you retrigger or merge anyway?

@ckamm can you please rebase? thanks

**Comment by ckamm** [comment]
@behzadnouri I fixed the runtime test I had overlooked.

**Comment by behzadnouri** [comment]
> @behzadnouri I fixed the runtime test I had overlooked.

thank you

**Pull request merged**